awx/main/tasks/facts.py

```diff
@@ -25,7 +25,8 @@
 def start_fact_cache(hosts, destination, log_data, timeout=None, inventory_id=None):
```

Comment:
Right now, `destination` is passed into this method which is like: private_data_dir/artifacts/42/fact_cache/

However, with this change, some stuff needs to be written at the folder starting at: private_data_dir/artifacts/42/

So I would start trying to be way more explicit here. Instead of `destination`, take `artifacts_dir`. And then for some parts of the code you need the fact_cache dir, so best to save that to a descriptive var like `fact_cache_dir = os.path.join(artifacts_dir, 'fact_cache')` when you come to that point.

Implication is that the _caller_ need to change, and when that happens, it's far better to pass it as a kwarg, like `artifacts_dir=`, because reading the task code is very hard as it is, and making the method signature clear helps.

Reply:
I believe I have accomplished this as well now!